### DISCUSSION TITLE
BLtouch not working correctly

### DISCUSSION THREAD

**Discussion by Akujiproxy** [Jyers/Marlin#1176] [Q&A] [answered]
Hi,

I recently compiled the firmware for my Ender 3 V2, using the source code and the BLTouch-3x3-HighSpeed Ender 3 V2 config template. The only thing I changed is that I enabled the Z_MIN_PROBE_USES_Z_MIN_ENDSTOP_PIN value as I have the hardware configured that way. 

However after leveling the bed manually and getting it relatively decently honed in, I then set the Z height and proceed to create a mesh. When the manually editing the mesh, all but the right bottom corner seem to be completely off. For instance, the mesh shows the left bottom has a positive value of 0.18 and there is just no way this is correct as the paper scratches during manual leveling. 

Anybody have an idea why the probe has these incredibly inaccurate results? Is there a value I overlooked that should also be changed when you change to the Z_MIN_PROBE_USES_Z_MIN_ENDSTOP_PIN method? I've been troubleshooting for days but cannot seem to find whats causing the issue..

More info in this reddit post:

https://www.reddit.com/r/FixMyPrint/comments/oo5tgc/bed_level_messed_up_by_abl/

**Comment by Akujiproxy** [comment]
I'm starting to feel like ABL has an inverted mesh. As if it exacerbates the bed leveling issues, rather than compensates for them.

**Reply by HermanGaviriaMarin** [reply]
Lol I thought the same thing too. I'm not relying on it compensating anymore. I create a mesh using UBL, then I autotilt until the mesh is as close to 0 as possible all the way around. Of course there are points rhat will never get to zero, but the closer the better and that should do the trick.

**Reply by Akujiproxy** [reply]
If I autotilt, the mesh just looks the same everywhere. So I feel that doesn't work?

**Comment by Jyers** [comment, marked as answer]
Hopefully you got it working eventually. If not maybe try the latest source :)
